plugins/usd/module/vtkF3DUSDImporter.cxx

```diff
@@ -0,0 +1,966 @@
+  {
+    if (!occlusionImage && !roughnessImage && !metallicImage)
+    {
+      return nullptr;
+    }
+
+    int maxWidth = 0;
+    int maxHeight = 0;
+
+    if (occlusionImage)
+    {
+      int* size = occlusionImage->GetDimensions();
+
+      maxWidth = std::max(maxWidth, size[0]);
+      maxHeight = std::max(maxHeight, size[1]);
+    }
+
+    if (roughnessImage)
+    {
+      int* size = roughnessImage->GetDimensions();
+
+      maxWidth = std::max(maxWidth, size[0]);
+      maxHeight = std::max(maxHeight, size[1]);
+    }
+
+    if (metallicImage)
+    {
+      int* size = metallicImage->GetDimensions();
+
+      maxWidth = std::max(maxWidth, size[0]);
+      maxHeight = std::max(maxHeight, size[1]);
+    }
+
+    auto ResizeAndExtractChannel = [&](vtkImageData* img) -> vtkSmartPointer<vtkImageData> {
+      if (!img)
+      {
+        vtkNew<vtkImageData> emptyImage;
+        emptyImage->SetDimensions(maxWidth, maxHeight, 1);
+        emptyImage->AllocateScalars(VTK_UNSIGNED_CHAR, 1);
+        unsigned char* data = static_cast<unsigned char*>(emptyImage->GetScalarPointer());
+        std::fill(data, data + maxWidth * maxHeight, 0xff);
+        return emptyImage;
+      }
+
+      // The image should already be a single channel image, but just in case
+      vtkNew<vtkImageExtractComponents> extract;
+      extract->SetInputData(img);
+      extract->SetComponents(0);
+
+      vtkNew<vtkImageResize> resize;
+      resize->SetInputConnection(extract->GetOutputPort());
+      resize->SetOutputDimensions(maxWidth, maxHeight, 1);
+      resize->Update();
+      return resize->GetOutput();
+    };
+
+    vtkNew<vtkImageAppendComponents> appendChannels;
+    appendChannels->SetInputData(ResizeAndExtractChannel(occlusionImage));
+    appendChannels->AddInputData(ResizeAndExtractChannel(roughnessImage));
+    appendChannels->AddInputData(ResizeAndExtractChannel(metallicImage));
+    appendChannels->Update();
+
+    return appendChannels->GetOutput();
+  }
+
+  vtkSmartPointer<vtkImageData> CombineColorOpacityImage(
+    vtkImageData* colorImage, vtkImageData* opacityImage)
+  {
+    if (!opacityImage)
+    {
+      return colorImage;
+    }
+
+    vtkNew<vtkImageAppendComponents> appendChannels;
+    appendChannels->SetInputData(colorImage);
+    appendChannels->AddInputData(opacityImage);
+    appendChannels->Update();
+
+    return appendChannels->GetOutput();
+  }
+
+  vtkSmartPointer<vtkImageData> GetVTKTexture(
+    const pxr::UsdShadeShader& samplerPrim, const pxr::TfToken& token)
+  {
+    if (!samplerPrim)
+    {
+      return nullptr;
+    }
+
+    auto& tex = this->TextureMap[samplerPrim.GetPath().GetAsString()];
+
+    if (tex == nullptr)
+    {
+      pxr::TfToken idToken;
+      bool defined = samplerPrim.GetIdAttr().Get(&idToken);
+      if (!defined || idToken != pxr::TfToken("UsdUVTexture"))
+      {
+        // only pxr::UsdUVTexture supported for now
+        return nullptr;
+      }
```

Comment:
should be easy to test with an unsuported texture type ?

Reply:
I think it's the most common texture type, if not the only one. The library is designed for future evolution so there are a lot of "token" like that.